sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsBuffer.scala

```diff
@@ -245,12 +245,24 @@ trait RapidsBuffer extends AutoCloseable {
   def getCopyIterator: RapidsBufferCopyIterator =
     new RapidsBufferCopyIterator(this)
 
+  /**
+   * At spill time, the tier we are spilling to may need to hand the rapids buffer an output stream
+   * to write to. This is the case for a `RapidsHostColumnarBatch`.
+   * @param outputStream stream that the `RapidsBuffer` will serialize itself to
+   */
+  def serializeToStream(outputStream: OutputStream): Unit = {
+    throw new IllegalStateException(s"Buffer $this does not support serializeToStream")
```

Comment:
Not a fan of APIs that throw in some cases when otherwise it could be avoided.  I'd rather see an API that something like a ByteChannel to serialize, and all RAPIDS host buffers implement it.  If they're just a HostMemoryBuffer then it's easy, just dump the bytes in the buffer to the channel.  If it's a host column batch, then it can wrap the channel in an output stream and use the jcudf approach.  Then we don't need a flag like `needsSerialization` to check whether some method is callable or not -- it's always callable.

Reply:
One issue is that we need a way to deserialize the host batch after it has gone to disk as a `RapidsDiskBuffer`. The `RapidsDiskBuffer` was using `needsSerialization` as a way to figure out "ah, this is not a regular buffer..` and I do need to carry this information around. I could create a `RapidsDiskHostColumnarBatch` which would be a subclass of `RapidsDiskBuffer` and just implements the deserialization side of things... open to ideas.

Reply:
https://github.com/NVIDIA/spark-rapids/pull/9098/commits/ff6ab4c1c2318e392589da06f566b1dded7511c1 should address this and other API concerns.